test/widgets/cards_test.dart

```diff
@@ -0,0 +1,344 @@
+import '../util/factories/ride_factory.dart';
+import '../util/mock_server.dart';
+import '../util/pump_material.dart';
+import '../util/request_processor.dart';
+import '../util/request_processor.mocks.dart';
+
+void main() {
+  final MockRequestProcessor processor = MockRequestProcessor();
+  late Ride ride;
+  late Drive drive;
+
+  setUpAll(() {
+    MockServer.setProcessor(processor);
+  });
+
+  setUp(() {
+    reset(processor);
+  });
+
+  //since the code is the same for RideCard DriveCard and PendingRideCard, we only test DriveCard
+  testWidgets('TripCard shows the correct information', (WidgetTester tester) async {
+    drive = DriveFactory().generateFake();
+    whenRequest(processor).thenReturnJson(drive.toJsonForApi());
+    await pumpMaterial(tester, DriveCard(drive, key: const Key('driveCard')));
+
+    //wait for card to load
+    await tester.pump();
+
+    //date is shown
+    expect(find.text(localeManager.formatDate(ride.startTime)), findsOneWidget);
+
+    //duration is shown
+    expect(find.byIcon(Icons.access_time_outlined), findsOneWidget);
+
+    //start location is shown
+    expect(find.byKey(const Key('start')), findsOneWidget);
+
+    //end location is shown
+    expect(find.byKey(const Key('end')), findsOneWidget);
+  });
+
+  group('PendingRideCard', () {
+    ride = RideFactory().generateFake(
+      status: RideStatus.pending,
+      rider: NullableParameter(ProfileFactory().generateFake(id: 1)),
+      seats: 1,
+    );
+    drive = DriveFactory().generateFake(rides: [ride], seats: 2);
+
+    setUp(() {
+      whenRequest(processor).thenReturnJson([]);
+    });
+
+    testWidgets('shows the correct information', (WidgetTester tester) async {
+      await pumpMaterial(tester, PendingRideCard(ride, reloadPage: () {}, drive: drive));
+
+      //Profile Widget from rider is shown
+      final Finder profileWidgetFinder = find.byType(ProfileWidget);
+      expect(profileWidgetFinder, findsOneWidget);
+      final ProfileWidget profileWidget = tester.widget(profileWidgetFinder);
+      expect(profileWidget.profile, ride.rider);
+
+      //Extra time is shown
+      expect(find.byKey(const Key('extraTime')), findsOneWidget);
+
+      //Seats indicator is shown
+      final Finder seatsIndicator = find.byType(IconWidget);
+      expect(seatsIndicator, findsOneWidget);
+      expect(find.descendant(of: seatsIndicator, matching: find.byIcon(Icons.chair)), findsOneWidget);
+
+      //Price is shown
+      expect(find.byKey(const Key('price')), findsOneWidget);
+
+      //Approve button is shown
+      expect(find.byKey(const Key('approveButton')), findsOneWidget);
+
+      //Reject button is shown
+      expect(find.byKey(const Key('rejectButton')), findsOneWidget);
+    });
+
+    group('approve Ride', () {
+      Future<void> openApproveDialog(WidgetTester tester) async {
+        //need scaffold for dialog
+        await pumpMaterial(tester, Scaffold(body: PendingRideCard(ride, reloadPage: () {}, drive: drive)));
```

Comment:
tbh maybe we should just add a Scaffold by default to `pumpMaterial`

Reply:
not sure since for the pages there already is a scaffold i think then it would be better to just add either a parameter `hasScaffold` or something or just use two methods

Reply:
Yes but when we write Widget tests that are not pages, we will always need this

Reply:
Or a `pumpScaffold` helper?

Reply:
yes that's what i meant. should i add it in this PR?

Reply:
Let's do it :D